pilot/pkg/model/validation.go

```diff
@@ -38,8 +38,8 @@ func (s *Service) Validate() error {
 	}
 	parts := strings.Split(string(s.Hostname), ".")
 	for _, part := range parts {
-		if !labels.IsDNS1123Label(part) {
-			errs = multierror.Append(errs, fmt.Errorf("invalid hostname part: %q", part))
+		if ok, err := labels.IsDNS1123Label(part); !ok {
+			errs = multierror.Append(errs, fmt.Errorf("invalid hostname part: %q, reason: %v", part, err))
```

Comment:
I don't object to "reason:", but typically in Go the error prefixes the underlying error.  Consider `invalid hostname part %q: %v` instead.  Consider this for all of your messages.